site/src/contexts/ProxyContext.tsx

```diff
@@ -6,25 +6,60 @@ import {
   createContext,
   FC,
   PropsWithChildren,
+  useCallback,
   useContext,
+  useEffect,
   useState,
 } from "react"
 import { ProxyLatencyReport, useProxyLatency } from "./useProxyLatency"
 
-interface ProxyContextValue {
+export interface ProxyContextValue {
+  // proxy is **always** the workspace proxy that should be used.
+  // The 'proxy.selectedProxy' field is the proxy being used and comes from either:
+  //   1. The user manually selected this proxy. (saved to local storage)
+  //   2. The default proxy auto selected because:
```

Comment:
is the default proxy the one with the lowest latency?

Reply:
Yes. I use the word `default`, maybe I should say `preferred`.

This proxy comes from this function: https://github.com/coder/coder/blob/stevenmasley%2Fproxy_latency_pick/site/src/contexts/ProxyContext.tsx#L218

It is based on lowest latency, but if the lowest latency information is not provided, then it is always the `primary` proxy.

---

I handle a lot of edge cases in this function that might never reasonably happen in production.